src/include/access/heaptoast.h

```diff
@@ -160,25 +124,6 @@ extern MemTuple toast_insert_or_update_memtup(Relation rel,
  */
 extern void toast_delete(Relation rel, HeapTuple oldtup, bool is_speculative);
 
-/* ----------
-<<<<<<< HEAD:src/include/access/tuptoaster.h
- * toast_delete_datum -
- *
- *	Delete a single external stored value.
- * ----------
- */
-extern void toast_delete_datum(Relation rel, Datum value, bool is_speculative);
-
-/* ----------
- * heap_tuple_fetch_attr() -
- *
- *		Fetches an external stored attribute from the toast
- *		relation. Does NOT decompress it, if stored external
- *		in compressed format.
- * ----------
- */
-extern struct varlena *heap_tuple_fetch_attr(struct varlena *attr);
```

Comment:
Why these functions also have been removed?

Reply:
They are not removed, but moved to access/toast_internals.h.

Reply:
Where? These are gpdb-specific functions, the upstream commit didn't touch them.

Reply:
> Commit https://github.com/arenadata/gpdb/commit/8b94dab06617ef80a0901ab103ebd8754427ef5a added declaration of toast_delete_datum to
src/include/access/toast_internals.h.

Reply:
> Commit https://github.com/arenadata/gpdb/commit/8b94dab06617ef80a0901ab103ebd8754427ef5a moved declarations of heap_tuple_fetch_attr,
heap_tuple_untoast_attr, and heap_tuple_untoast_attr_slice from
src/include/access/heaptoast.h to src/include/access/detoast.h

Reply:
only `toast_delete_datum` and `varattrib_untoast_len` are gpdb-specific